File: src/audio_stegano_app/lib/core/audio/audio_load_errors.dart
import '../../app/app_strings.dart';

/// User-facing message when [AudioInputLoader] or decoders fail.
String audioLoadErrorMessage(AppStrings strings, Object error) {
  final text = error.toString();
  if (text.contains('MP4 decode failed')) {
    return strings.errorMp4Decode;
  }
  if (text.contains('MP3 decode failed') ||
      text.contains('Media decode failed')) {
    return strings.errorMp3Decode;
  }
  return text;
}

File: src/audio_stegano_desktop/src/AudioStegano.Desktop/Localization/AudioLoadErrors.cs
namespace AudioStegano.Desktop.Localization;

/// <summary>User-facing audio load/decode errors (Flutter <c>audio_load_errors.dart</c> parity).</summary>
public static class AudioLoadErrors
{
    public static string Format(AppStrings strings, object error)
    {
        var text = error.ToString() ?? string.Empty;
        if (text.Contains("MP4 decode failed", StringComparison.Ordinal))
            return strings.ErrorMp4Decode;

        if (text.Contains("MP3 decode failed", StringComparison.Ordinal) ||
            text.Contains("Media decode failed", StringComparison.Ordinal))
            return strings.ErrorMp3Decode;

        return text;
    }
}
